operator/pkg/manifest/shared.go

```diff
@@ -490,9 +493,44 @@ func unmarshalAndValidateIOP(iopsYAML string, force, allowUnknownField bool, l c
 		l.LogAndError("Run the command with the --force flag if you want to ignore the validation error and proceed.")
 		return iop, fmt.Errorf(errs.Error())
 	}
+
 	return iop, nil
 }
 
+// resolveConflictSettings When user uses configuration file or --set flag to change value for parameters, in some
```

Comment:
Even though we handle PDB as a special case, I don't think it's a good idea to put handling here, this will change the IOP spec, users may complain about that.
Instead, I would like put it to k8s overlay after helm rendering, maybe here: https://github.com/istio/istio/blob/14561dd76e1cacca6685cbd915d84e1d037b6030/operator/pkg/translate/translate.go#L173

Reply:
@morvencao Will look into that and see if that is not too late.

Reply:
i'm not sure there's a huge difference where we change IOP as long as it's done internally as in either case it's opaque to the user. the real question is - is this a good idea at all? we had a long discussion about that and fundamentally there's no way to address this problem without masking values coming from default profiles. the best we could do is put it into a framework to give this a feel of respectability, but that seems like an overkill for just one parameter. if we need to do this type of fixup for other paths, we should consider it then.